File: src/main/java/model/pieces/Bishop.java
package model.pieces;

import model.moves.Move;
import model.util.Colour;
import model.util.Offset;
import org.jetbrains.annotations.NotNull;

public class Bishop extends DirectionPiece {
    private static final Offset[] DIRECTIONS = {
            Offset.TOP_LEFT,
            Offset.TOP_RIGHT,
            Offset.BOTTOM_LEFT,
            Offset.BOTTOM_RIGHT
    };


    public Bishop(Colour colour) {
        super(colour);
    }

    @NotNull
    @Override
    Offset[] getDirections() {
        return DIRECTIONS;
    }

    @Override
    int getUnicodeWhite() {
        return 9815;
    }

    @Override
    int getUnicodeBlack() {
        return 9821;
    }

    @Override
    public int getUnsignedValue() {
        return 3;
    }

    @Override
    public void notifyMoveComplete(Move move) {
    }

    @Override
    public void notifyMoveUndo(Move move) {
    }

    @NotNull
    @Override
    String getName() {
        return "Bishop";
    }
}

File: chess-engine/src/main/java/model/pieces/Bishop.kt
package model.pieces

import model.moves.Move
import model.util.Colour
import model.util.Offset

internal class Bishop(colour: Colour) : DirectionPiece(colour) {

    override val directions: Array<Offset>
        get() = DIRECTIONS

    override val unicodeWhite: Int
        get() = 9815

    override val unicodeBlack: Int
        get() = 9821

    override val unsignedValue: Int
        get() = 3

    override val name: String
        get() = "Bishop"

    override fun notifyMoveComplete(move: Move) {}

    override fun notifyMoveUndo(move: Move) {}

    companion object {
        private val DIRECTIONS = arrayOf(Offset.TOP_LEFT, Offset.TOP_RIGHT, Offset.BOTTOM_LEFT, Offset.BOTTOM_RIGHT)
    }
}
